lib/debug/tracer.rb

```diff
@@ -58,14 +58,20 @@ def skip? tp
       if tp.path.start_with?(__dir__) ||
          tp.path.start_with?('<internal:') ||
          ThreadClient.current.management? ||
-         (@pattern && !tp.path.match?(@pattern) && !tp.method_id.match?(@pattern)) ||
```

Comment:
Exception event's `method_id` could be `nil` and thus cause `NoMethodError` when called with `match?`

Reply:
`method_id&.` is enough, and `||` is needed. I'll fix it later.

Reply:
I missed `||` in refactor 🤦 
I've added https://github.com/ruby/debug/pull/228 for the fix.